core/controllers/android_e2e_config.py

```diff
@@ -0,0 +1,301 @@
+
+            This is only called once and raises a exception if the topic is
+            already created or published.
+
+        Raises:
+            Exception. When used in production mode.
+            Exception. The topic is already created but not published.
+            Exception. The topic is already published.
+        """
+
+        if not constants.DEV_MODE:
+            raise Exception('Cannot load new structures data in production.')
+        if topic_services.does_topic_with_name_exist(
+                'Android test'):
+            topic = topic_fetchers.get_topic_by_name('Android test')
+            topic_rights = topic_fetchers.get_topic_rights(
+                topic.id, strict=False)
+            if topic_rights.topic_is_published:
+                raise Exception('The topic is already published.')
+            else:
+                raise Exception('The topic exists but is not published.')
+        exp_id = '26'
+        user_id = feconf.SYSTEM_COMMITTER_ID
+        # Generate new Structure id for topic, story, skill and question.
+        topic_id = topic_fetchers.get_new_topic_id()
+        story_id = story_services.get_new_story_id()
+        skill_id = skill_services.get_new_skill_id()
+        question_id = question_services.get_new_question_id()
+
+        # Create dummy skill and question.
+        skill = self._create_dummy_skill(
+            skill_id, 'Dummy Skill for Android', '<p>Dummy Explanation 1</p>')
+        question = self._create_dummy_question(
+            question_id, 'Question 1', [skill_id])
+        question_services.add_question(user_id, question)
+        question_services.create_new_question_skill_link(
+            user_id, question_id, skill_id, 0.3)
+
+        # Create and update topic to validate before publishing.
+        topic = topic_domain.Topic.create_default_topic(
+            topic_id, 'Android test', 'test-topic-one', 'description')
+        topic.update_url_fragment('test-topic')
+        topic.update_meta_tag_content('tag')
+        topic.update_page_title_fragment_for_web('page title for topic')
+        topic.update_thumbnail_filename('test_svg.svg')
+        topic.update_thumbnail_bg_color('#C6DCDA')
+
+        # Add other structures to the topic.
+        topic.add_canonical_story(story_id)
+        topic.add_uncategorized_skill_id(skill_id)
+        topic.add_subtopic(1, 'Test Subtopic Title')
+
+        # Update and validate subtopic.
+        topic.update_subtopic_thumbnail_filename(1, 'test_svg.svg')
+        topic.update_subtopic_thumbnail_bg_color(1, '#FFFFFF')
+        topic.update_subtopic_url_fragment(1, 'suburl')
+        topic.move_skill_id_to_subtopic(None, 1, skill_id)
+        subtopic_page = (
+            subtopic_page_domain.SubtopicPage.create_default_subtopic_page(
+                1, topic_id))
+
+        # Upload local exploration to the datastore and enable feedback.
+        exp_services.load_demo(python_utils.convert_to_bytes(
+            exp_id))
```

Comment:
Why do we need `python_utils.convert_to_bytes` here?

Reply:
Removed, not needed

Reply:
Just checked, it is not needed here.